includes/class-wc-payments-utils.php

```diff
@@ -591,10 +591,16 @@ public static function get_filtered_error_message( Exception $e ) {
 	 * @return  int
 	 */
 	public static function get_filtered_error_status_code( Exception $e ) : int {
+		$status_code = 400;
 		if ( $e instanceof API_Exception ) {
-			return $e->get_http_code() ?? 400;
+			$status_code = $e->get_http_code() ?? 400;
+			// Sometimes hosting companies hijack this status code and return their own predefined error page.
+			// In this case, we want to return a 400 instead, since it will break the flow of the checkout page.
+			if ( 402 === $status_code ) {
+				$status_code = 400;
+			}
```

Comment:
@zmaglica do you plan to add a test for this change? As I checked the [test file](https://github.com/Automattic/woocommerce-payments/blob/6f59377afd8fb5ebe175bc565e240d69475ea899/tests/unit/test-class-wc-payments-utils.php) does exist.

Reply:
> @zmaglica do you plan to add a test for this change? As I checked the [test file](https://github.com/Automattic/woocommerce-payments/blob/6f59377afd8fb5ebe175bc565e240d69475ea899/tests/unit/test-class-wc-payments-utils.php) does exist.

I am not planning to write tests in this PR for this file, since the tests for this particular file don't exist, as you mentioned too. I will keep this PR as clean as possible and open a new issue to address the issue with the tests.

Reply:
@zmaglica I meant the test file, which is [this](https://github.com/Automattic/woocommerce-payments/blob/6f59377afd8fb5ebe175bc565e240d69475ea899/tests/unit/test-class-wc-payments-utils.php), does exist :))

IMO, it is always a good idea to write test in the same PR.